### ISSUE TITLE
DirectPlay a > 1Mbps video stream even if Internet streaming limit set to 1 Mbps.

### ISSUE THREAD

**Issue by Elaws** [jellyfin/jellyfin-android#564]
### Describe the bug

1. Set Internet streaming limit to 1 Mbs.
2. Jellyfin should transcode if video stream exceeds 1 Mbps.
3. Jellyfin does not transcode, and direcplays the video. Thus, video is constantly stuttering.

### Application version

2.4.0

### Where did you install the app from?

Google Play

### Device information

Samsung Galaxy S20FE 5G

### Android version

Android 11

### Jellyfin server version

10.7.7

### Which video player implementations does this bug apply to?

- [ ] Web player (default)
- [X] Integrated player (ExoPlayer)
- [ ] External player (VLC, mpv, MX Player)

**Comment by Elaws** [comment]
Well, it seems the problem happens only with ExoPlayer, not with web player. 

For unknown reasons, any internet speed limit or quality settings that you can set in the app is absolutely not at all taken into account when using ExoPlayer. No transcoding occurs when you set limits.

**Comment by Maxr1998** [comment]
Yes, known issue, I mentioned it the other day in #563. The tracking issue for this is also #24 (in addition to user-selectable bitrate limits), some work has already been done on  #535.

**Comment by naimo** [comment]
By the way, the exoplayer (and I believe the external player option as well, with VLC for example) also ignores the server "Internet streaming bitrate limit"
If that helps

**Comment by galmok** [comment]
This issue is still a problem and it is exacerbated by the fact the the web player doesn't work (freeze frame unless constantly touched). As such, there exists no player on Android with bitrate limiting (transcoding) applied.

**Comment by Elaws** [comment]
I can confirm that the issue is still happening. Has there been any progress concerning this issue reported 17 months ago ?

**Comment by extrange** [comment]
This is still happening with Jellyfin on Android, with both the integrated player and external player (VLC).

**Comment by Maxr1998** [comment]
ExoPlayer now supports manual bitrate limits. The user provided limits will be supported in the future.

**Comment by shaino** [comment]
Actually, it's not difficult.
in
app/src/main/java/org/jellyfin/mobile/player/interaction/playoptions.kt
line 134,add code... for example:
viewModal.changeBitrate(6000000)

But
To be precise, it is necessary to obtain locally set parameters to assist.

**Comment by shaino** [comment]
If the developer has not yet completed it, they can use it urgently:
step 1 
In file: 
app/src/main/java/org/jellyfin/mobile/player/ui/PlayerFragment.kt
add code in fun onCreate(savedInstanceState: Bundle?) line 135
            var currentExoBitrate:Int = 4000000
            val cachPath  = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).absolutePath
            val fileName = cachPath + File.separator + "jellyexoplayer.rate"
            val cfgFile = File(fileName)
            if(cfgFile.exists()){
                val rateStrin = File(fileName).readText()
                currentExoBitrate = rateStrin.toInt()
            }
            if(currentExoBitrate > 120000000 || currentExoBitrate < 0)
            {
                currentExoBitrate = 0
            }
            viewModel.changeBitrate(currentExoBitrate)

step 2
in file:
app/src/main/java/org/jellyfin/mobile/player/playerViewModel.kt
add code in fun chageBitrate(bitrate: Int?): Boolean line 444
  	val cachPath  = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).absolutePath
        val fileName = cachPath + File.separator + "jellyexoplayer.rate"
        val cfgFile = File(fileName)
        File(fileName).writeText(bitrate.toString())
Note: Add before the following line
        return queueManager.changBitrate(bitrate)

Of course, both kt files need to be added to import
import java.io.File
import android.os.Environment

Wishing everyone a pleasant day

**Comment by nielsvanvelzen** [comment]
Feel free to open a pull request

**Comment by J3m5** [comment]
Are there any updates on this issue?

**Comment by Maxr1998** [comment]
Unfortunately not. I'm still in the middle of the player UI rewrite without much progress recently since I have been pretty busy.

**Comment by xDaveN** [comment]
Been a year, and I've just run into this, lol

**Comment by sebastianrasor** [comment]
Just another user dealing with this issue

**Comment by kelthor85** [comment]
Hi, for as long as I can remember, I've had to limit the max bitrate to force transcoding then select hardware decoding to play anything that isn't 1080p x264.  It's my fault for using an ancient tablet, but still, it would be great not having to tap the screen and change 2 settings for every single file.  Web view has an incredible amount of frame drops, unfortunately, otherwise I would've just used that. 

Thank you kindly.

**Comment by cani1see** [comment]
this bug is so old, why not make a persist just for exoplayer, so that at least one config is respected?
